File: monkey.py
from random import randint

# python 版猴子问题

length = 20
apes = []

for i in range(0,length):
    apes.append(20)

def god(min,max,exclude):
    i = randint(min,max)
    while i == exclude :
        i = randint(min,max)
    return i

for i in range(0,365 * 20):
    for j in range(0,length):
        if apes[j] != 0:
            apes[j] -= 1
            apes[god(0,length-1,j)] += 1

apes.sort()
for i in range(0,length):
    print('-' * apes[i],apes[i])

File: monkey.java
import java.util.Arrays;
import java.util.Random;

/**
 * Java版猴子问题
 */
class Monkey {
    public static void main(String[] args) {
        int[] apes = new int[20];

        for (int i = 0; i < apes.length; i++) {
            apes[i] = 20;
        }

        for (int i = 0; i < 365 * 20; i++) {
            for (int j = 0; j < apes.length; j++) {
                if (apes[j] != 0) {
                    apes[j] -= 1;
                    apes[god(apes.length - 1, j)] += 1;
                }
            }
        }

        Arrays.sort(apes);

        for (int ape : apes) {
            System.out.println("-".repeat(ape) + ape);
        }
    }

    public static int god(int max, int exclude) {
        int temp;
        Random random = new Random();
        do {
            temp = random.nextInt(max);
        } while (temp == exclude);
        return temp;
    }
}